src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -247,20 +275,62 @@ public void RemoveItems(IEnumerable collection)
         /// <returns>IDisposable.</returns>
         public IDisposable SuspendChangeNotifications()
         {
-            return new DisposableToken<FastObservableCollection<T>>(this,
+            return SuspendChangeNotifications(SuspensionMode.Mixed);
+        }
+
+        /// <summary>
+        /// Suspends the change notifications until the returned <see cref="IDisposable"/> is disposed.
+        /// <example>
+        /// <code>
+        /// <![CDATA[
+        /// var fastCollection = new FastObservableCollection<int>();
+        /// using (fastCollection.SuspendChangeNotificaftions())
+        /// {
+        ///     // Adding or removing events inside here will not raise events
+        ///     fastCollection.Add(1);
+        ///     fastCollection.Add(2);
+        ///     fastCollection.Add(3);
+        /// 
+        ///     fastCollection.Remove(3);
+        ///     fastCollection.Remove(2);
+        ///     fastCollection.Remove(1);
+        /// }
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="mode">The suspension Mode.</param>
+        /// <returns>IDisposable.</returns>
+        public IDisposable SuspendChangeNotifications(SuspensionMode mode)
+        {
+            if (_suspendChangeNotifications && this._suspensionContext.Mode != mode)
```

Comment:
I cannot. Do you understand why I did this respectively what the problem is?
If I remove this e.x. InsertItem(int index, T item) will throw the exception.